Vienna/Sources/Main window/FoldersTree.m

```diff
@@ -1601,4 +1290,311 @@ -(void)dealloc
+            }
+        }
+        
+        // If parent was a group, expand it now
+        if (parentId != VNAFolderTypeRoot)
+            [self.outlineView expandItem:[self.rootNode nodeFromID:parentId]];
+        
+        // Select a new folder
+        if (folderToSelect > 0)
+            [self selectFolder:folderToSelect];
+        
+        return YES;
+    }
+    return NO;
+}
+
+-(NSView *)outlineView:(NSOutlineView *)outlineView viewForTableColumn:(NSTableColumn *)tableColumn item:(id)item {
+    FolderTreeCellView *folderTreeCellView = nil;
+    if ([tableColumn.identifier isEqualToString:@"folderColumns"])
+    {
+        folderTreeCellView = [outlineView makeViewWithIdentifier:@"FolderTreeCell" owner:self];
+        TreeNode * node = (TreeNode *)item;
+        if (node == nil) {
+            node = self.rootNode;
+        }
+        
+        static NSDictionary * info = nil;
+        if (info == nil)
+        {
+            NSMutableParagraphStyle * style = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
+            style.lineBreakMode = NSLineBreakByTruncatingTail;
+            style.tighteningFactorForTruncation = 0.0;
+        style.alignment = NSTextAlignmentLeft;
+            info = @{NSParagraphStyleAttributeName: style};
+        }
+        
+        Folder * folder = node.folder;
+        NSMutableDictionary * myInfo = [NSMutableDictionary dictionaryWithDictionary:info];
+        if (folder.isUnsubscribed) {
+                myInfo[NSForegroundColorAttributeName] = NSColor.secondaryLabelColor;
+            } else {
+                myInfo[NSForegroundColorAttributeName] = NSColor.labelColor;
+        }
+        // Set the font
+        if (folder.unreadCount ||  (folder.childUnreadCount && ![outlineView isItemExpanded:item])) {
+            myInfo[NSFontAttributeName] = self.boldCellFont;
+        }
+        else {
+            myInfo[NSFontAttributeName] = self.cellFont;
+        }
+        
+        // Only show folder images if the user prefers them.
+        Preferences * prefs = [Preferences standardPreferences];
+        folderTreeCellView.imageView.image = (prefs.showFolderImages ? folder.image : [folder standardImage]);
+        folderTreeCellView.textField.attributedStringValue = [[NSAttributedString alloc] initWithString:node.nodeName attributes:myInfo];
+        folderTreeCellView.textField.delegate = self;
+        
+        // Use the auxiliary position of the feed item to show
+        // the refresh icon if the feed is being refreshed, or an
+        // error icon if the feed failed to refresh last time.
+        if (folder.isUpdating) {
+            folderTreeCellView.inProgress = YES;
+        }
+        else if (folder.isError)
+        {
+            folderTreeCellView.showError = YES;
+            folderTreeCellView.inProgress = NO;
+        }
+        else
+        {
+            folderTreeCellView.showError = NO;
+            folderTreeCellView.inProgress = NO;
+        }
```

Comment:
Make brace position consistent. But even better, you could write it as two lines of boolean statements:
folderTreeCellView.inProgress = folder.isUpdating
and
folderTreeCellView.showError = !folder.isUpdating && folder.isError